lib/primitives/block.js

```diff
@@ -880,7 +904,8 @@ class Block extends AbstractBlock {
       }
     }
 
-    return GCSFilter.fromItems(19, key, items);
+    const params=this.buildFilterParams(filterType);
+    return GCSFilter.fromItems(params.P, params.M, key, items);
```

Comment:
I think this section may need a second thought because the spec for BASIC is not just the two GCS constants [but also the contents of the filter itself](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki#contents) 

A future filter type may use the same parameters even but encode different content (like maybe INCLUDING OP_RETURN, or something like taproot annexes)

So we should discuss, but maybe `toFilter(view, filterType)` has a simple switch statement and if the type is BASIC, we delegate completely to another function `toBasicFilter()` where the constants and transaction elements are just hard coded, as defined in BIP158?